### PR TITLE
MDEV-14578: mysql_install_db install unix_socket plugin when --auth-r…

### PR DESCRIPTION
**Pull request by grooverdan**
Needed to check if aria existed in bootstrap otherwise we couldn't check the plugin existed (MDEV-16255).

I submit this under the MCA.

### CODE REVIEW

File: scripts/mysql_install_db.sh [hunk 1]
```diff
@@ -307,6 +307,7 @@ then
   langdir="$basedir/sql/share/english"
   srcpkgdatadir="$srcdir/scripts"
   buildpkgdatadir="$builddir/scripts"
+  plugindir="$builddir/plugin/auth_socket/"
```
Comment (svoj) [inline scripts/mysql_install_db.sh]: I have no idea how to solve it better, but :(

File: scripts/mysql_system_tables_data.sql [hunk 2]
```diff
@@ -46,8 +46,13 @@ INSERT INTO tmp_user_nopasswd VALUES ('localhost','root','','Y','Y','Y','Y','Y',
 REPLACE INTO tmp_user_nopasswd SELECT @current_hostname,'root','','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','','','','',0,0,0,0,'','','N','N','',0 FROM dual WHERE @current_hostname != 'localhost';
 REPLACE INTO tmp_user_nopasswd VALUES ('127.0.0.1','root','','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','','','','',0,0,0,0,'','','N','N','',0);
 REPLACE INTO tmp_user_nopasswd VALUES ('::1','root','','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','','','','',0,0,0,0,'','','N','N', '', 0);
--- More secure root account using unix sucket auth.
+-- More secure root account using unix socket auth.
 INSERT INTO tmp_user_socket VALUES ('localhost',IFNULL(@auth_root_socket, 'root'),'','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','Y','','','','',0,0,0,0,'unix_socket','','N', 'N','', 0);
+-- Need aria support to lookup information_schema.plugins (result is tmptable)
```
Comment (svoj) [inline scripts/mysql_system_tables_data.sql]: But then, don't you need aria to lookup `INFORMATION_SCHEMA.ENGINES`? :)

**Comment by svoj** [comment]
Hi @grooverdan,

Thanks for your contribution! I wonder if it works well with mtr? It is quite picky about system tables contents.

Thanks,
Sergey

File: scripts/mysql_system_tables_data.sql (hunk 2 above)
Comment (grooverdan) [reply scripts/mysql_system_tables_data.sql]: Apparently not, I removed ```IF @auth_root_socket is not null THEN``` and ran mtr which does --loose-skip-aria and it still succeeded. Maybe ```ENGINES`` has an index. I'm not going to spoil the mystery by looking.

**Comment by grooverdan** [comment]
mtr was the biggest pain. It doesn't set @auth_root_socket so its from an empty test. This is the reason its a nested ```IF``` rather than ```AND`` operators. Running mtr against an installed instance would hit the same pickyness regardless of this change.

File: scripts/mysql_system_tables_data.sql (hunk 2 above)
Comment (vuvova) [reply scripts/mysql_system_tables_data.sql]: In my tests you don't need Aria to test for unix_socket either. At least, I clearly see in the debugger that it creates a MEMORY table, not Aria.

Comment (grooverdan) [reply scripts/mysql_system_tables_data.sql]: Quite right, removed. I still had it there from the lightly different expression I had in MDEV-16255.

**Comment by vuvova** [comment]
merged in 10.2, thanks

**Comment by vuvova** [comment]
Hmm, I didn't see that, I tried a script few times in a row and there were no errors

**Pull request closed without merging**